feat: add React useState lab (#635)
freeCodeCamp/CurriculumExpansion: frontend-cert/react-projects/react-use-state/user-stories.md

> 1. You should use a `useState` hook and destruct the assignment to two elements returned- `bgColor`(current state value) and `setBgColor`(function that you can call to update the bgColor state).

Reviewer: ```suggestion
1. You should use the `useState` hook and destructure the two elements it returns: `bgColor` (the current state value) and `setBgColor` (the function used to update the `bgColor` state).
```

> 2. The background colour of the page should initally be set to white using the `useState` hook.

Reviewer: all instances of color should use American english instead of british

3. The `#pick-color-btn` button should have the `onClick` property set to the `handleButtonClick` function.

> 4. The `handleButtonClick` function should trigger a click event on the `#color-input` button to open the color picker.

Reviewer: also, this user story can be removed since this function is no longer used
